src/spicy/runtime-support.cc

```diff
@@ -221,6 +221,18 @@ TypePtr rt::event_arg_type(const EventHandlerPtr& handler, const hilti::rt::inte
     return zeek_args[idx];
 }
 
+zeek::analyzer::Analyzer* rt::current_analyzer() {
```

Comment:
As the other comment, there's file and packet analyzers, too. either `current_protocol_analyzer()`, or maybe `current_analyzer_id()` that happens to work for protocol analyzers and return `0` otherwise?

But probably @rsmmr  should mull this over :-)

Reply:
➕ for making this work for all kinds. With that change most of the `spicy` test pass, but 4 still fail since some events don't seem to get raised. This might not be a purely Spicy problem anymore though.